journal/202211.md

```diff
@@ -0,0 +1,514 @@
+- [Enabled](https://github.com/decred/dcrd/pull/3017) and set [as default](https://github.com/decred/dcrd/pull/3019) new peer-to-peer protocol version that no longer uses the `reject` message. The message was inherited from the original btcsuite codebase and was used to signal that a transaction or a block is ["rejected"](https://en.bitcoin.it/wiki/Protocol_documentation#reject) by a peer. Using this message for pretty much anything was unreliable or incorrect and a [proposal](https://github.com/decred/dcrd/issues/2546) was made to remove it. Peers negotiating to this new protocol version will no longer send `reject` messages and consider it a violation if they receive it. Any peers negotiating to an older version will not be punished for sending `reject` and it will be ignored by newer nodes.
+- Added new [IPC events](https://github.com/decred/dcrd/pull/3020) to notify the parent process of the address and port assigned to dcrd. In configurations where dcrd is controlled by another process like Decrediton or an automated test runner, that parent process needs to know which addresses and ports dcrd uses for its P2P interface (talking to the public network) and RPC interface (local communications with wallets and other clients). This change provides a reliable way to discover dcrd's ports when they are randomly assigned by the operating system and not configured by the parent process.
+- Always respond to [`getheaders`](https://github.com/decred/dcrd/pull/3030) requests from peers. Previous logic would not respond to `getheaders` if the local chain was not considered to be fully synced with the network. In the old "blocks first" syncing model it helped to avoid some undesirable corner cases, like unnecessarily downloading blocks. It had downsides too, like a node could appear to be unresponsive or stalled to its peers, while being active and simply not having any interesting data to share. In the current "headers first" syncing model this limitation is no longer necessary, and serving up headers before the peer is known to be current no longer leads to undesirable behavior.
+- Other internal/infra changes { 3021 3023 3022 - drop after review }
+
+RPC server:
+
+- Optimized WebSocket client disconnection handling and fixed spurious messages when WebSocket clients disconnect.
+- Modified the [lifecycle](https://github.com/decred/dcrd/pull/3025) of WebSocket clients to use contexts, which is more flexible for the calling code.
+- Added [cancellation support](https://github.com/decred/dcrd/pull/3027) for the `getwork` request (used by miners) and optimized handling of multiple concurrent `getwork` requests.
+  - { _could not understand deeper what blocked what before this cange, or what setups experienced a bottleneck. maybe current text is enough but I can add more if it's fun_ }
```

Comment:
I think the current text is fine. as it gets pretty technical to delve in any further and it really only potentially affected miners who don't play nicely with the API to begin with and mostly notably around initial startup where it can take a while to reach the initial state sync prior to usable work becoming available.

Moreover, it only applies to miners who are using polling via `getwork` to obtain work, and, irrespective of this change, miners are are encouraged to use the Decred-specific asynchronous notifications of new work (`work`) provided by `notifywork` instead of the more common polling based approach that virtually every other coin uses.

Reply:
Agreed to not delve any further but this is interesting stuff so instead I've used this bullet to add a bit of general education. Thanks for explaining.